source/nodejs/adaptivecards-designer/src/designer-peer-treeitem.ts

```diff
@@ -58,4 +59,28 @@ export class DesignerPeerTreeItem extends BaseTreeItem {
     getChildAt(index: number): BaseTreeItem {
         return this.owner.getChildAt(index).treeItem;
     }
+
+    protected click(e: MouseEvent) {
+        super.click(e);
+
+        let currentPeer = this.owner;
+
+        while (currentPeer) {
+            if (currentPeer instanceof CarouselPagePeer) {
```

Comment:
I don't think `DesignerPeerTreeItem` shoul have logic that depends on any specific peer type. That logic should be implemented in the specific peer class.

Here, what you want is when the peer for a carousel page is clicked then that page should be made visible. This extends to all peers: when a peer is selected, its associated card element should be brought into view. So what you should do is add a `bringCardElementIntoView(): boolean` method to `CardElementPeer` that returns `false` (meaning "I didn't do anything") in the base implementation. You ovveride it in `CarouselPagePeer` with the logic you currently have here, and you return `true` when you have actually slid to another page. And here, you can change the logic to simply:
```typescript
while (currentPeer) {
    if (currentPeer.bringIntoView()) {
        break;
    }

    currentPeer = currentPeer.parent;
}
```

Reply:
Just incorporated this change :)